RadixWallet/Features/PreAuthorizationReview/PreAuthorizationReview+View.swift

```diff
@@ -0,0 +1,268 @@
+				store.send(.view(.copyRawManifestButtonTapped))
+			} toggleAction: {
+				store.send(.view(.toggleDisplayModeButtonTapped))
+			}
+		}
+
+		private func details(_ showRawManifestButton: Bool) -> some SwiftUI.View {
+			sections
+				.padding(.top, .large2 + .small3)
+				.padding(.horizontal, .small1)
+				.padding(.bottom, .medium1)
+				.overlay(alignment: .topTrailing) {
+					if showRawManifestButton {
+						Button(asset: AssetResource.code) {
+							store.send(.view(.toggleDisplayModeButtonTapped))
+						}
+						.buttonStyle(.secondaryRectangular)
+						.padding(.medium3)
+					}
+				}
+				.frame(minHeight: .standardButtonHeight + 2 * .medium3, alignment: .top)
+		}
+
+		private var sections: some SwiftUI.View {
+			let childStore = store.scope(state: \.sections, action: \.child.sections)
+			return Common.Sections.View(store: childStore)
+		}
+
+		private func feesInformation(dAppName: String?) -> some SwiftUI.View {
+			HStack(spacing: .zero) {
+				VStack(alignment: .leading, spacing: .zero) {
+					Text(L10n.PreAuthorizationReview.Fees.title(dAppName ?? "dApp"))
+						.foregroundStyle(.app.gray1)
+
+					Text(L10n.PreAuthorizationReview.Fees.subtitle)
+						.foregroundStyle(.app.gray2)
+				}
+				.lineSpacing(0)
+				.textStyle(.body2Regular)
+
+				Spacer(minLength: .small2)
+
+				InfoButton(.preauthorizations)
+			}
+			.padding(.vertical, .medium3)
+			.padding(.horizontal, .medium2)
+			.background(Color.app.gray5)
+			.clipShape(RoundedRectangle(cornerRadius: .small1))
+		}
+
+		@ViewBuilder
+		private func expiration(_ expiration: Expiration?, secondsToExpiration: Int?) -> some SwiftUI.View {
+			Group {
+				switch expiration {
+				case .atTime:
+					if let seconds = secondsToExpiration {
+						if seconds > 0 {
+							let value = formatTime(seconds: seconds)
+							Text(markdown: L10n.PreAuthorizationReview.Expiration.atTime(value), emphasizedColor: .app.account4pink, emphasizedFont: .app.body2Link)
+						} else {
+							Text(L10n.PreAuthorizationReview.Expiration.expired)
+						}
+					}
+
+				case let .afterDelay(value):
+					let value = formatTime(seconds: Int(value.expireAfterSeconds))
+					Text(markdown: L10n.PreAuthorizationReview.Expiration.afterDelay(value), emphasizedColor: .app.account4pink, emphasizedFont: .app.body2Link)
+
+				case nil:
+					Color.clear
+				}
+			}
+			.textStyle(.body2Regular)
+			.foregroundStyle(.app.account4pink)
+			.padding(.horizontal, .medium1)
+			.frame(minHeight: .huge2)
+		}
+	}
+}
+
+private extension PreAuthorizationReview.View {
+	/// Given an amount of seconds, returns a formatted String using the corresponding unit (days/hours/minutes/seconds).
+	/// A few examples on how should it look for each of them:
+	/// - `8 days` / `1 day`
+	/// - `23:21 hours` / `1:24 hour`
+	/// - `56:02 minutes` / `1:23 minute`
+	/// - `34 seconds` / `1 second`
+	func formatTime(seconds: Int) -> String {
+		typealias S = L10n.PreAuthorizationReview.TimeFormat
```

Comment:
I would probably not have this formatting logic inside a view, can we move it to Sargon and expose a function?

Reply:
or just move it from view layer and add some unit tests?